File: moveZeroes.cpp
    class Solution {
    public:
        void moveZeroes(vector<int>& nums) {
            // brute
            // vector<int> temp(nums.size(), 0);
            // int index = 0;

            // for(int i=0; i<nums.size(); i++){
            //     if(nums[i]!=0) {
            //         temp[index] = nums[i];
            //         index++;
            //     }
            // }

            // for(int i=0; i<nums.size(); i++){
            //     nums[i]=temp[i];
            // }
            
    // optimal

            int j=-1;

            for(int i=0; i<nums.size(); i++){
                if(nums[i]==0){
                    j=i;
                    break;
                }
            }

            if(j==-1) return;

            for(int i=j+1; i<nums.size(); i++){
                if(nums[i]!=0) {
                    swap(nums[i], nums[j]);
                    j++;
                }
            }
        }

    };

File: moveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # brute

        # temp = [0] * len(nums)

        # index = 0

        # for i in nums:
        #     if i != 0:
        #         temp[index]=i
        #         index+=1

        # for i in range(len(nums)):
        #     nums[i]=temp[i]

        # optimal

        j=-1

        for i in range(len(nums)):
            if nums[i] == 0:
                j=i
                break

        if j==-1:
            return

        for i in range(j+1, len(nums)):
            if nums[i] !=0:
                nums[i], nums[j] = nums[j], nums[i]
                j+=1

        


        